File: program/program/find-the-largest-three-elements-in-an-array/find_the_largest_three_elements_in_an_array.dart
import 'dart:io';

void main() {
  print("Enter the size of the array:");
  int size = int.parse(stdin.readLineSync()!);

  List<int> arr = [];

  for (int i = 0; i < size; i++) {
    print("Enter element ${i + 1}:");
    int element = int.parse(stdin.readLineSync()!);
    arr.add(element);
  }

  List<int> largestThree = findLargestThreeElements(arr);

  print("The largest three elements in the array are:");
  for (int i = 0; i < largestThree.length; i++) {
     
    print("${i + 1}: ${largestThree[i]}");
    print(" ");
  }
}

List<int> findLargestThreeElements(List<int> arr) {
  arr.sort((a, b) => b.compareTo(a)); // Sort the array in descending order
  List<int> largestThree = [];

  for (int i = 0; i < 3 && i < arr.length; i++) {
    largestThree.add(arr[i]);
  }

  return largestThree;
}

File: program/program/find-the-largest-three-elements-in-an-array/find_the_largest_three_elements_in_an_array.rs
use std::io;

fn main() {
    println!("Enter the size of the array:");
    let mut size = String::new();
    io::stdin().read_line(&mut size).expect("Failed to read line");
    let size: usize = size.trim().parse().expect("Invalid input");
    let mut numbers = Vec::new();

    println!("Enter {} elements:", size);
    for _ in 0..size {
        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        let number: i32 = input.trim().parse().expect("Invalid input");
        numbers.push(number);
    }
    numbers.sort_by(|a, b| b.cmp(a));

 
    println!("The largest three elements are:");
    for i in 0..3 {
        println!("{}", numbers[i]);
    }
}
